src/Utilities/penalty_relaxation.jl

```diff
@@ -0,0 +1,239 @@
+# Copyright (c) 2017: Miles Lubin and contributors
+# Copyright (c) 2017: Google Inc.
+#
+# Use of this source code is governed by an MIT-style license that can be found
+# in the LICENSE.md file or at https://opensource.org/licenses/MIT.
+
+"""
+    PenaltyRelaxation(
+        penalties = Dict{MOI.ConstraintIndex,Float64}();
+        default::Union{Nothing,T} = 1.0,
+    )
+
+A problem modifier that, when passed to [`MOI.modify`](@ref), destructively
+modifies the model in-place to create a penalized relaxation of the constraints.
+
+!!! warning
+    This is a destructive routine that modifies the model in-place. If you don't
+    want to modify the original model, use `JuMP.copy_model` to create a copy
+    before setting this attribute.
+
+## Reformulation
+
+The penalty relaxation modifies constraints of the form ``f(x) \\in S`` into
+``f(x) + y - z \\in S``, where ``y, z \\ge 0``, and then it introduces a penalty
+term into the objective of ``a \\times (y + z)`` (if minimizing, else ``-a``),
+where `a` is the value in the `penalties` dictionary associated with the
+constraint that is being relaxed. If no value exists, the default is `default`.
```

Comment:
This does not say what happens if `default` is `nothing`

Reply:
Done